src/Federation/FederationTypeInterceptor.cs

```diff
@@ -295,4 +300,128 @@ private void CompleteReferenceResolver(ObjectTypeDefinition typeDef)
         objectTypeDefinition.Directives.Add(
             new DirectiveDefinition(directiveNode));
     }
+
+    private void AddInaccessibleMarkers(DefinitionBase? definition)
```

Comment:
this is a pretty ugly workaround as currently `HotChocolate` does not allow you to create directive attributes that would be applicable to multiple targets, since `@tag`/`@inaccessible` are technically applicable on `FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ENUM | ENUM_VALUE | SCALAR | INPUT_OBJECT | INPUT_FIELD_DEFINITION | ARGUMENT_DEFINITION` we would need separate attribute per each possible location....